SRVCOM-1272 Report UI tests in junit format

mgencur opened the pull request.

cardil commented on Makefile:
```diff
@@ -50,7 +50,7 @@ test-e2e:
 
 # Run E2E tests from the current repo for serving+eventing+knativeKafka
-test-e2e-with-kafka:
+test-e2e-with-kafka: test-ui-e2e
```
This is temporal change, right?

/hold

mgencur replied:
Right. No need to hold. This PR is WIP

mgencur commented:
The previous run correctly reports the failed tests as junit: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/874/pull-ci-openshift-knative-serverless-operator-main-4.7-operator-e2e-aws-ocp-47/1379427360650039296
I've just squashed the changes and removed the temporary commit. The UI tests won't run in pre-submits now.

mgencur commented:
/hold cancel

cardil commented:
/assign

cardil commented:
The primary goal of reporting all UI tests as xUnit is achieved, so I can give lgtm.

I don't like the results are also printed on console as bloody XML. That's ugly. I rather, like to see a nice human-readable output like default `spec` reporter. That would involve using multiple reporters like in this example: https://docs.cypress.io/guides/tooling/reporters#Spec-to-STDOUT-save-JUnit-XML-files. We can do it in follow-up PR, if you like...

mgencur commented:
/hold

mgencur commented:
Removed the temporary commit. The PR is ready for another review.

cardil reviewed (approved):
Looks great. Thanks.

/lgtm

mgencur commented:
/hold cancel
/retest

cardil commented:
/retest
